### PR TITLE
Favor  more C++-like lifecycle cleanup. over xmalloc/xfree 

### PR DESCRIPTION
**Pull request by GPSBabelDeveloper**
Open old worm cans to reduce the number of copies and byte counts of C strings/buffer transitions, manual memory management, and other "old school" programming in several formats that aren't exactly strategic to us.

### CODE REVIEW

**Comment by tsteven4** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Coverage increased per file
===========================
+ garmin_gpi.cc  1
+ alan.cc  1
         

Coverage decreased per file
===========================
- garmin_tables.cc  -7
- vitosmt.cc  -2
- dmtlog.cc  -1
         

Clones removed
==============
+ gtm.cc  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/gpsbabel/gpsbabel/pullRequest?prid=5637348&bid=18462372)

File: duplicate.cc [hunk 1]
```diff
@@ -152,7 +152,7 @@ void DuplicateFilter::process()
 
   int ct = waypt_count();
 
-  auto* htable = (wpt_ptr*) xmalloc(ct * sizeof(wpt_ptr));
+  wpt_ptr htable[ct];
```
Comment (tsteven4) [inline duplicate.cc:155]: I don't think this is legal in C++.  It was legal in C99.  g++ allows this as an extension:
https://gcc.gnu.org/onlinedocs/gcc/Variable-Length.html
but MSVC does not.

My preference would be with stick with the standard.

Comment (tsteven4) [reply duplicate.cc:155]: QScopedArrayPointer works fairly nice here, although it can require some changes to how the pointer is used.

**Comment by robertlipe** [comment]
I think all the good stuff from this PR has been merged on other PRs. Abandoning.

**Pull request closed without merging**